crates/ruff_linter/resources/test/fixtures/pylint/unnecessary_assignment.py

```diff
@@ -0,0 +1,78 @@
+bad1 = 'example'
+if bad1: # [consider-using-assignment-expr]
+    pass
+
+bad2 = 'example'
+if bad2 and True: # [consider-using-assignment-expr]
+    pass
+
+bad3 = 'example'
+if bad3 and bad3 == 'example': # [consider-using-assignment-expr]
+    pass
+
+
+def foo():
+    bad4 = 0
+    if bad4: # [consider-using-assignment-expr]
+        pass
+
+bad5 = (
+    'example',
+    'example',
+    'example',
+    'example',
+    'example',
+    'example',
+    'example',
+    'example',
+    'example',
+    'example',
+)
+if bad5: # [consider-using-assignment-expr]
+    pass
+
+bad6 = 'example'
+if bad6 is not None: # [consider-using-assignment-expr]
+    pass
+
+bad7 = 'example'
+if bad7 == 'something': # [consider-using-assignment-expr]
+    pass
+elif bad7 == 'something else':
+    pass
```

Comment:
This is an interesting example and possibly controversial. I would prefer the existing solution because the assignment in the `if` is very subtle and I can see how it can be confusing when trying to figure out what the value of `bad7` is in the `elif` branch